cmd/kubebuilder/initproject/vendor.go

```diff
@@ -17,172 +17,250 @@ limitations under the License.
-
-			// Delete the directory if it is a repo or package in a repo
-			if dir != "vendor" && parent != "vendor" && !(gparent == "vendor" && parent == "github.com") {
-				os.RemoveAll(filepath.Dir(p))
-			}
-		}
+	util.Write(depManifestFile, "dep-manifest-file", depManifestTmpl, depTmplArgs)
+	if err := runDepEnsure(); err != nil {
+		fmt.Printf("Error running 'dep ensure': %s\n", err)
 	}
+	return
 }
 
-func RunVendorInstall(cmd *cobra.Command, args []string) {
-	fmt.Printf("\t%s/\n", filepath.Join("vendor"))
-
-	// Delete old versions of the packages we manage before installing the new ones
-	if Update {
-		deleteOld()
-	}
-
-	// Get the executable directory
-	e, err := os.Executable()
+func runDepEnsure() error {
+	fmt.Printf("Updating vendor dependencies. Running 'dep ensure'....\n")
+	cmd := exec.Command("dep", "ensure")
+	o, err := cmd.CombinedOutput()
 	if err != nil {
-		log.Fatal("unable to get directory of kubebuilder tools")
+		fmt.Printf("Failed to run 'dep ensure': %s\n", string(o))
+		return err
 	}
-	e = filepath.Dir(e)
+	fmt.Printf("Updated vendor dependencies successfully.\n")
+	return nil
+}
 
-	// read the file
-	f := filepath.Join(e, "vendor.tar.gz")
-	fr, err := os.Open(f)
+// TODO(droot): may be move to util fn ?
+func copyFile(src, dst string) error {
+	from, err := os.Open(src)
 	if err != nil {
-		log.Fatalf("failed to read vendor tar file %s %v", f, err)
+		return err
 	}
-	defer fr.Close()
+	defer from.Close()
 
-	// setup gzip of tar
-	gr, err := gzip.NewReader(fr)
+	to, err := os.OpenFile(dst, os.O_RDWR|os.O_CREATE, 0666)
 	if err != nil {
-		log.Fatalf("failed to read vendor tar file %s %v", f, err)
-	}
-	defer gr.Close()
-
-	// setup tar reader
-	tr := tar.NewReader(gr)
-
-	for file, err := tr.Next(); err == nil; file, err = tr.Next() {
-		if file.FileInfo().IsDir() {
-			continue
-		}
-		p := filepath.Join(".", file.Name)
-		if Update && filepath.Dir(p) == "." {
-			continue
-		}
-
-		err := os.MkdirAll(filepath.Dir(p), 0700)
-		if err != nil {
-			log.Fatalf("Could not create directory %s: %v", filepath.Dir(p), err)
-		}
-		b, err := ioutil.ReadAll(tr)
-		if err != nil {
-			log.Fatalf("Could not read file %s: %v", file.Name, err)
-		}
-		err = ioutil.WriteFile(p, b, os.FileMode(file.Mode))
-		if err != nil {
-			log.Fatalf("Could not write file %s: %v", p, err)
-		}
+		return err
 	}
+	defer to.Close()
 
-	err = updateDepConfig()
+	_, err = io.Copy(to, from)
 	if err != nil {
-		log.Fatalf("Could not update Gopkg.toml file: %v", err)
+		return err
 	}
+	return nil
 }
 
-// updateDepConfig updates the Dep config Gopkg.toml to include Kubebuilder
-// project dependency. It uses the Kubebuilder version to determine whether
-// to include branch or version in the contraint stanza.
-func updateDepConfig() error {
-	var depConstraint string
+// template for dep's manifest file (Gopkg.toml). This is generated using
+// scripts/generate_dep_manifest.sh scripts.
```

Comment:
Just want to make things clear. If we update the dependency of kubebuilder itself, do we need to rerun the script `generate_dep_manifest.sh` to regenerate this template? So that the dep in generated projects is in sync with kubebuilder itself.

Reply:
Good question.  Maybe we need a script to wrap dep ensure?